docs: change visibility in transactions (#5245)
ydb-platform/ydb: ydb/docs/en/core/concepts/_includes/transactions.md

* {{ ydb-short-name }} does not support transactions that combine DDL and DML queries. The conventional [ACID]{% if lang == "en" %}(https://en.wikipedia.org/wiki/ACID){% endif %}{% if lang == "ru" %}(https://ru.wikipedia.org/wiki/ACID){% endif %}  notion of a transactions is applicable specifically to DML queries, that is, queries that change data. DDL queries must be idempotent, meaning repeatable if an error occurs. If you need to manipulate a schema, each manipulation is transactional, while a set of manipulations is not.
* YQL implementation used in {{ ydb-short-name }} employs the [Optimistic Concurrency Control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) mechanism. If an entity is affected during a transaction, optimistic blocking is applied. When the transaction is complete, the mechanism verifies that the locks have not been invalidated. For the user, locking optimism means that when transactions are competing with one another, the one that finishes first wins. Competing transactions fail with the `Transaction locks invalidated` error.

> * All changes made during the transaction accumulate in the database server memory and are applied when the transaction completes. If the locks are not invalidated, all the changes accumulated are committed atomically; if at least one lock is invalidated, none of the changes are committed. The above model involves certain restrictions: changes made by a single transaction must fit inside the available memory.

Reviewer: This is not correct anymore, a transaction can write data to a DS persistently, gigabytes, publish it after commit

Author: > This is not correct anymore, a transaction can write data to a DS persistently, gigabytes, publish it after commit

Talked to the developers, and this limitation is still with us. Reference links:
- https://t.me/c/1254401711/6885
- https://t.me/c/1254401711/6889

I will retain the phrase where it is for now.

> For efficient execution, a transaction should be formed in a way that the first part of the transaction only reads data, while the second part of the transaction only changes data. The query structure then looks as the following:

Reviewer 2: ```suggestion
For efficient execution, a transaction should be formed so that the first part of the transaction only reads data, while the second part of the transaction only changes data. The query structure then looks as follows:
```
